test/e2e/apibinding/apibinding_webhook_test.go

```diff
@@ -0,0 +1,334 @@
+	err = v1.AddToScheme(scheme)
+	require.NoError(t, err, "failed to add admission v1 scheme")
+	err = v1alpha1.AddToScheme(scheme)
+	require.NoError(t, err, "failed to add cowboy v1alpha1 to scheme")
+	cowbyClients, err := client.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to add cowboy v1alpha1 to scheme")
+	codecs := serializer.NewCodecFactory(scheme)
+	deserializer := codecs.UniversalDeserializer()
+
+	//Create Test Server and Create Validating Webhook for Cowboys in source cluster.
+	testWebhook := webhookserver.WebhookServer{
+		Response: v1.AdmissionResponse{
+			Allowed: true,
+		},
+		ObjectGVK: schema.GroupVersionKind{
+			Group:   "wildwest.dev",
+			Version: "v1alpha1",
+			Kind:    "Cowboy",
+		},
+		T:            t,
+		Lock:         sync.Mutex{},
+		Deserializer: deserializer,
+	}
+
+	port, err := framework.GetFreePort(t)
+	require.NoError(t, err, "failed to get free port for test webhook")
+	testWebhook.StartServer(ctx, server, port)
+
+	t.Logf("Installing webhook into the source workspace")
+	sideEffect := admissionregistrationv1.SideEffectClassNone
+	webhook := &admissionregistrationv1.MutatingWebhookConfiguration{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{Name: "test-webhook"},
+		Webhooks: []admissionregistrationv1.MutatingWebhook{{
+			Name: "test-webhook.cowboy.io",
+			ClientConfig: admissionregistrationv1.WebhookClientConfig{
+				URL:      testWebhook.GetURL(),
+				CABundle: cfg.CAData,
+			},
+			Rules: []admissionregistrationv1.RuleWithOperations{{
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+				},
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{"wildwest.dev"},
+					APIVersions: []string{"v1alpha1"},
+					Resources:   []string{"cowboys"},
+				},
+			}},
+			SideEffects:             &sideEffect,
+			AdmissionReviewVersions: []string{"v1"},
+		}},
+	}
+	_, err = kubeClusterClient.Cluster(sourceWorkspace).AdmissionregistrationV1().MutatingWebhookConfigurations().Create(ctx, webhook, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to add validating webhook configurations")
+
+	cowboy := v1alpha1.Cowboy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testing",
+		},
+		Spec: v1alpha1.CowboySpec{},
+	}
+
+	// Avoid race condition here by making sure that CRD is served after installing the types into logical clusters
+	t.Logf("Creating cowboy resource in target logical cluster")
+	require.Eventually(t, func() bool {
+		_, err = cowbyClients.Cluster(targetWorkspace).WildwestV1alpha1().Cowboys("default").Create(ctx, &cowboy, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return false
+		}
+		return true
+
+	}, wait.ForeverTestTimeout, 100*time.Millisecond)
+
+	//Avoid race condition where webhook informer is not updated before the call to create was made.
+	t.Log("Verify webhook is eventually called")
+	require.Eventually(t, func() bool {
+		return testWebhook.Calls == 1
```

Comment:
would also like to see a check that the in-workspace webhook (to be installed) is not called.